packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -374,6 +375,24 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
   /* @conditional-compile-remove(pinned-participants) */
   const [pinnedParticipantsState, setPinnedParticipantsState] = React.useState<string[]>([]);
   /* @conditional-compile-remove(pinned-participants) */
+  const [selectedScalingModeState, setselectedScalingModeState] = React.useState<{ [key: string]: VideoStreamOptions }>(
+    {}
+  );
+
+  /* @conditional-compile-remove(pinned-participants) */
+  const updateScalingModeForRemoteUser = useCallback(
+    (remoteUserId: string, scalingMode: ViewScalingMode) => {
+      setselectedScalingModeState((current) => ({
+        ...current,
+        [remoteUserId]: {
+          scalingMode,
+          isMirrored: remoteVideoViewOptions?.isMirrored
+        }
+      }));
+    },
+    [remoteVideoViewOptions]
```

Comment:
```suggestion
    [remoteVideoViewOptions?.isMirrored]
```

So we don't depend on an object but a boolean value, so no updated will be needed most of time